File: test_cafekiosk/src/main/java/com/testcafekiosk/spring/domain/product/ProductRepository.java
//package com.testcafekiosk.spring.domain.product;
//
//import org.springframework.data.jpa.repository.JpaRepository;
//
//import java.util.List;
//
//public interface ProductRepository extends JpaRepository<Product, Long> {
//
//    /**
//     * select *
//     * from product
//     * where selling_status in ('SELLING', 'HOLD');
//     */
//    List<Product> findAllBySellingStatusIn(List<ProductSellingType> sellingStatuses);
//    List<Product> findAllByProductNumberIn(List<String> productNumbers);
//}

File: test_cafekiosk/src/main/kotlin/com/testcafekiosk/spring/domain/product/ProductRepository.kt
package com.testcafekiosk.spring.domain.product

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query

interface ProductRepository : JpaRepository<Product, Long> {

    fun findAllBySellingStatusIn(sellingStatus: List<ProductSellingStatus>): List<Product>

    fun findAllByProductNumberIn(productNumbers: List<String>): List<Product>

    @Query("select p.productNumber from Product p " +
            "order by p.id desc " +
            "limit 1")
    fun findLatestProductNumber(): String?
}